### PR TITLE
Add extension method for easier `ResourceManagementOptions` registration (Lombiq Technologies: OCORE-208)

### PR DESCRIPTION
**Pull request by Piedone**
Following the pattern of all the `Add*` extensions.

The implementation is in `src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs`, and the release notes are the notable changes, the rest is just search and replace.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -14,4 +15,14 @@ public static IServiceCollection AddResourceManagement(this IServiceCollection s
...
         return services;
     }
+
+    /// <summary>
+    /// Adds a service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to the service collection.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of the service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to.
+    /// </typeparam>
+    public static IServiceCollection AddResourceManagementOptionsConfiguration<T>(this IServiceCollection services)
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs:25]: Seb hates @MikeAlhayek for introducing such extensions, seems it's your turn you :)

Comment (hishamco) [inline src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs:25]:
```suggestion
    public static IServiceCollection ConfigureResourceManagementOptions<T>(this IServiceCollection services)
```

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Startup.cs [hunk 2]
```diff
@@ -25,7 +23,7 @@ public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Startup.cs:26]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.ContentPreview/Startup.cs [hunk 3]
```diff
@@ -9,15 +8,14 @@
...
-using OrchardCore.ResourceManagement;
 
 namespace OrchardCore.ContentPreview;
 
 public sealed class Startup : Modules.StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentPreview/Startup.cs:18]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/Startup.cs [hunk 4]
```diff
@@ -37,7 +35,7 @@ public override void ConfigureServices(IServiceCollection services)
         services.AddRecipeExecutionStep<DeleteContentDefinitionStep>();
 
         services.AddTransient<IRecipeEventHandler, LuceneRecipeEventHandler>();
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/Startup.cs:38]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Facebook/Startup.cs [hunk 5]
```diff
@@ -33,7 +32,7 @@ public override void ConfigureServices(IServiceCollection services)
         services.AddSiteDisplayDriver<FacebookSettingsDisplayDriver>();
         services.AddRecipeExecutionStep<FacebookSettingsStep>();
 
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Facebook/Startup.cs:35]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Flows/Startup.cs [hunk 6]
```diff
@@ -48,6 +46,6 @@ public override void ConfigureServices(IServiceCollection services)
 
         services.AddDataMigration<Migrations>();
 
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Flows/Startup.cs:49]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs [hunk 7]
```diff
@@ -80,7 +78,7 @@ public override void ConfigureServices(IServiceCollection services)
         .AddLiquidFilter<ShapeRenderFilter>("shape_render")
         .AddLiquidFilter<ShapeStringifyFilter>("shape_stringify");
 
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs:81]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Media/Startup.cs [hunk 8]
```diff
@@ -85,7 +84,7 @@ public override void ConfigureServices(IServiceCollection services)
         .AddLiquidFilter<AssetUrlFilter>("asset_url")
         .AddLiquidFilter<ResizeUrlFilter>("resize_url");
 
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media/Startup.cs:87]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Notifications/Startup.cs [hunk 9]
```diff
@@ -66,7 +64,7 @@ public override void ConfigureServices(IServiceCollection services)
 
         services.Configure<NotificationOptions>(_shellConfiguration.GetSection("OrchardCore_Notifications"));
 
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, NotificationOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<NotificationOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Notifications/Startup.cs:67]:
```suggestion
        services.ConfigureResourceManagementOptions<NotificationOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Resources/Startup.cs [hunk 10]
```diff
@@ -31,7 +30,7 @@ public override void ConfigureServices(IServiceCollection serviceCollection)
             o.LiquidViewParserConfiguration.Add(parser => parser.RegisterParserBlock("styleblock", parser.ArgumentsListParser, StyleBlock.WriteToAsync));
         });
 
-        serviceCollection.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        serviceCollection.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Resources/Startup.cs:33]:
```suggestion
        serviceCollection.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Spatial/Startup.cs [hunk 11]
```diff
@@ -19,7 +17,7 @@ public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Spatial/Startup.cs:20]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Templates/Startup.cs [hunk 12]
```diff
@@ -19,7 +17,7 @@ public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Templates/Startup.cs:20]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Themes/Startup.cs [hunk 13]
```diff
@@ -24,7 +22,7 @@ public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Themes/Startup.cs:25]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.UrlRewriting/Startup.cs [hunk 14]
```diff
@@ -28,7 +26,7 @@ public override void ConfigureServices(IServiceCollection services)
         services.AddUrlRewritingServices()
             .AddNavigationProvider<AdminMenu>()
             .AddPermissionProvider<UrlRewritingPermissionProvider>()
-            .AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>()
+            .AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>()
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.UrlRewriting/Startup.cs:29]:
```suggestion
            .ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>()
```

File: src/OrchardCore.Modules/OrchardCore.Users/Startup.cs [hunk 15]
```diff
@@ -155,7 +154,7 @@ public override void ConfigureServices(IServiceCollection services)
         });
 
         services.AddTransient<IUsersAdminListFilterProvider, DefaultUsersAdminListFilterProvider>();
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, UserOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<UserOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/Startup.cs:157]:
```suggestion
        services.ConfigureResourceManagementOptions<UserOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Widgets/Startup.cs [hunk 16]
```diff
@@ -29,6 +27,6 @@ public override void ConfigureServices(IServiceCollection services)
         services.AddScoped<IContentTypePartDefinitionDisplayDriver, WidgetsListPartSettingsDisplayDriver>();
         services.AddContentPart<WidgetMetadata>();
         services.AddDataMigration<Migrations>();
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Widgets/Startup.cs:30]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Modules/OrchardCore.Workflows/Startup.cs [hunk 17]
```diff
@@ -85,7 +83,7 @@ public override void ConfigureServices(IServiceCollection services)
         services.AddActivity<LogTask, LogTaskDisplayDriver>();
 
         services.AddRecipeExecutionStep<WorkflowTypeStep>();
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Startup.cs:86]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Themes/TheAdmin/Startup.cs [hunk 18]
```diff
@@ -18,7 +16,7 @@ public Startup(IShellConfiguration configuration)
 
     public override void ConfigureServices(IServiceCollection services)
     {
-        services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Themes/TheAdmin/Startup.cs:19]:
```suggestion
        services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Themes/TheAgencyTheme/Startup.cs [hunk 19]
```diff
@@ -1,14 +1,12 @@
...
-using OrchardCore.ResourceManagement;
 
 namespace OrchardCore.Themes.TheAgencyTheme;
 
 public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection serviceCollection)
     {
-        serviceCollection.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        serviceCollection.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Themes/TheAgencyTheme/Startup.cs:10]:
```suggestion
        serviceCollection.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Themes/TheBlogTheme/Startup.cs [hunk 20]
```diff
@@ -1,14 +1,12 @@
...
-using OrchardCore.ResourceManagement;
 
 namespace OrchardCore.Themes.TheBlogTheme;
 
 public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection serviceCollection)
     {
-        serviceCollection.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        serviceCollection.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Themes/TheBlogTheme/Startup.cs:10]:
```suggestion
        serviceCollection.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/OrchardCore.Themes/TheComingSoonTheme/Startup.cs [hunk 21]
```diff
@@ -1,14 +1,12 @@
...
-using OrchardCore.ResourceManagement;
 
 namespace OrchardCore.Themes.TheComingSoonTheme;
 
 public sealed class Startup : StartupBase
 {
     public override void ConfigureServices(IServiceCollection serviceCollection)
     {
-        serviceCollection.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        serviceCollection.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/OrchardCore.Themes/TheComingSoonTheme/Startup.cs:10]:
```suggestion
        serviceCollection.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/docs/reference/modules/Resources/README.md [hunk 22]
```diff
@@ -142,7 +142,7 @@ In this example, we define a style that depends on Bootstrap version 4. In this
 
 !!! note "Registration"
     Make sure to register this `IConfigureOptions<ResourceManagementOptions>` in the `Startup` or your theme or module.
-    `serviceCollection.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();`
+    `serviceCollection.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();`
```
Comment (hishamco) [inline src/docs/reference/modules/Resources/README.md:145]:
```suggestion
    `serviceCollection.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();`
```

File: src/docs/releases/3.0.0.md [hunk 23]
```diff
@@ -107,3 +107,19 @@ will return `false` for administrators, even though they still have full access.
...
+When adding an `IConfigureOptions<ResourceManagementOptions>` implementation, used to add static resources and commonly named `ResourceManagementOptionsConfiguration`, you previously had to do the following in `Startup` classes:
+
+```csharp
+services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+```
+
+To simplify this, we introduced a new extension method to do the same in a shorter form:
+
+```csharp
+services.AddResourceManagementOptionsConfiguration<ResourceManagementOptionsConfiguration>();
```
Comment (hishamco) [inline src/docs/releases/3.0.0.md:122]:
```suggestion
services.ConfigureResourceManagementOptions<ResourceManagementOptionsConfiguration>();
```

File: src/docs/releases/3.0.0.md [hunk 24]
```diff
@@ -107,3 +107,19 @@ will return `false` for administrators, even though they still have full access.
...
+```
+
+You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceManagementOptionsConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```
Comment (hishamco) [inline src/docs/releases/3.0.0.md:125]:
```suggestion
You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `ConfigureResourceManagementOptions<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (Piedone) [reply src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs:25]: Just one suggestion for these is enough for me to get the idea ;). However, I disagree: all of these shortcuts we have are named `Add*`, and for a reason: we're adding a type to the service collection, following methods like `AddTransient`. The `Configure*` methods deal directly with options classes, e.g.:

```c#
services.Configure<TemplateOptions>(o =>
{
    o.MemberAccessStrategy.Register<DisplayMediaFieldViewModel>();
    o.MemberAccessStrategy.Register<Anchor>();

    o.Filters.AddFilter("img_tag", MediaFilters.ImgTag);
})
```

This rather adds a service that implements `IConfigureOptions<T>`, only which in turn deals with options classes.

Comment (hishamco) [reply src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs:25]: Internally you configure `ResourceManifest` by adding an entry on each call

Comment (Piedone) [reply src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs:25]: Yeah, but that's not what this method is doing. If we were to swap the method in this:

```c#
services.Configure<ResourceManagementOptions>(o =>
{
    _manifest = new ResourceManifest();
    _manifest
        .DefineScript("coming-soon")
        .SetUrl("~/TheComingSoonTheme/js/scripts.min.js", "TheComingSoonTheme/js/scripts.js")
        .SetVersion("6.0.5");

    _manifest
        .DefineStyle("coming-soon")
        .SetUrl("~/TheComingSoonTheme/css/styles.min.css", "TheComingSoonTheme/css/styles.css")
        .SetVersion("6.0.5");

    o.ResourceManifests.Add(_manifest);
})
```

Then we could name it `ConfigureResourceManagementOptions` and thus do:

```c#
services.ConfigureResourceManagementOptions(o =>
{
   ...
    o.ResourceManifests.Add(_manifest);
})
```

There the name would be appropriate. However, not for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, T>()` substitution I implemented.

**Comment by MikeAlhayek** [comment]
How about calling it AddResourceConfigurations?

**Comment by Piedone** [comment]
Even better, renamed.

File: src/docs/releases/3.0.0.md [hunk 25]
```diff
@@ -107,3 +107,19 @@ will return `false` for administrators, even though they still have full access.
...
+services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+```
+
+To simplify this, we introduced a new extension method to do the same in a shorter form:
+
+```csharp
+services.AddResourceConfiguration<ResourceManagementOptionsConfiguration>();
+```
+
+You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```
Comment (MikeAlhayek) [inline src/docs/releases/3.0.0.md:125]: Or you. An right click on IConfigureOptions and select show all references :)

File: src/docs/releases/3.0.0.md [hunk 26]
```diff
@@ -107,3 +107,19 @@ will return `false` for administrators, even though they still have full access.
 Many type commonly used by classes can be `sealed`, which improves runtime performance. While it's not mandatory, we recommend that you consider applying this improvement to your own extensions as well. We've implemented this enhancement in pull request [#16897](https://github.com/OrchardCMS/OrchardCore/pull/16897).
 
 ## Change Log
+
```
Comment (MikeAlhayek) [inline src/docs/releases/3.0.0.md:110]: I would add these under miscellaneous section or add these notes under "Resources" section.

File: src/docs/reference/modules/Resources/README.md [hunk 27]
```diff
@@ -142,7 +142,7 @@ In this example, we define a style that depends on Bootstrap version 4. In this
 
 !!! note "Registration"
     Make sure to register this `IConfigureOptions<ResourceManagementOptions>` in the `Startup` or your theme or module.
-    `serviceCollection.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();`
+    `serviceCollection.AddResourceConfiguration<ResourceManagementOptionsConfiguration>();`
```
Comment (MikeAlhayek) [inline src/docs/reference/modules/Resources/README.md:145]:
```suggestion
    `services.AddResourceConfiguration<ResourceManagementOptionsConfiguration>();`
```

File: src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs [hunk 28]
```diff
@@ -14,4 +15,14 @@ public static IServiceCollection AddResourceManagement(this IServiceCollection s
 
         return services;
     }
+
+    /// <summary>
+    /// Adds a service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to the service collection.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of the service implementing <see cref="IConfigureOptions{ResourceManagementOptions}"/> to.
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.ResourceManagement/ServiceCollectionExtensions.cs:23]:
```suggestion
    /// The type of the implementing of <see cref="IConfigureOptions{ResourceManagementOptions}"/> to register.
```

**Review by MikeAlhayek — APPROVED** [review]
Minor suggestions

File: src/docs/releases/3.0.0.md (hunk 25 above)
Comment (Piedone) [reply src/docs/releases/3.0.0.md:125]: That'll show you where the affected code lines are, but then you'd need to manually change them. With a regex (and obviously also checking the git diff, and testing the code) you get it done in 5 seconds :).

File: src/docs/releases/3.0.0.md (hunk 26 above)
Comment (Piedone) [reply src/docs/releases/3.0.0.md:110]: Created a Miscellaneous section.

**Comment by Piedone** [comment]
@hishamco anything else you'd like to add?

File: src/docs/releases/3.0.0.md [hunk 29]
```diff
@@ -107,3 +107,21 @@ will return `false` for administrators, even though they still have full access.
...
+## Miscellaneous
+
+### New Extension Method for Adding `IConfigureOptions<ResourceManagementOptions>` Implementations
+
+When adding an `IConfigureOptions<ResourceManagementOptions>` implementation, used to add static resources and commonly named `ResourceManagementOptionsConfiguration`, you previously had to do the following in `Startup` classes:
```
Comment (hishamco) [inline src/docs/releases/3.0.0.md:115]:
```suggestion
When adding an `IConfigureOptions<ResourceManagementOptions>` implementation, used to add static resources and commonly named `ResourceManagementOptionsConfiguration`, you previously had to do the following in the `Startup` classes:
```

File: src/docs/releases/3.0.0.md [hunk 30]
```diff
@@ -107,3 +107,21 @@ will return `false` for administrators, even though they still have full access.
...
+services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+```
+
+To simplify this, we introduced a new extension method to do the same in a shorter form:
+
+```csharp
+services.AddResourceConfiguration<ResourceManagementOptionsConfiguration>();
+```
+
+You can utilize this in your codebase by doing a search for the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```
Comment (hishamco) [inline src/docs/releases/3.0.0.md:127]:
```suggestion
You can utilize this in your codebase by searching the `AddTransient<IConfigureOptions<ResourceManagementOptions>, (.+)>\(\)` regex pattern and replacing it with `AddResourceConfiguration<$1>()`. Projects using this have to reference the `OrchardCore.ResourceManagement` package.
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by hishamco** [comment]
Just minor changes and ready to merge

**Comment by Piedone** [comment]
Thanks!

**Review by hishamco — APPROVED** [review]

**Pull request merged**